Userland/Libraries/LibWeb/Bindings/WindowObject.cpp

```diff
@@ -102,7 +102,18 @@ Origin WindowObject::origin() const
 
 static DOM::Window* impl_from(JS::VM& vm, JS::GlobalObject& global_object)
 {
-    auto* this_object = vm.this_value(global_object).to_object(global_object);
+    // Since this is a non built-in function we must treat it as non-strict mode
+    // this means that a nullish this_value should be converted to the
+    // global_object. Generally this does not matter as we try to convert the
+    // this_value to a specific object type in the bindings. But since window is
+    // the global object we make an exception here.
+    // This allows calls like `setTimeout(f, 10)` to work.
```

Comment:
I'm not completely happy with this comment but did not want to go on and on.
If you know of a better way to phrase this or some other things I should mention please say so.

Also do we maybe want a comment in VM.cpp about naive functions being strict if they are part of the ECMAScript spec?

Reply:
seems fine :)